core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala

```diff
@@ -37,9 +43,44 @@ case class IntelliJPatternResolver(pattern: String) extends IntelliJResolver {
       "version" -> version.releaseOrBuild,
       "release" -> version.release.getOrElse("snapshot-release")
     )
-    val replaced = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
+    val replacedBeforeResolvingGlobs = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
       path.replace(s"[$pattern]", replacement)
     }
+    val replaced = replaceGlobsWithExistingDirectories(List(replacedBeforeResolvingGlobs)).head
     Dependency(replaced)
   }
+
+  // solution below assumes that each * character is used to mark one part of the path (one atomic directory),
+  // for example: "file:///home/.cache/ides/com.jetbrains.intellij.idea/ideaIC/[revision]/*/ideaIC-[revision]/"
+  @tailrec
+  private def replaceGlobsWithExistingDirectories(paths: List[String]): List[String] =
+    if (paths.exists(pathMightBeValidResource))
+      paths.filter(pathMightBeValidResource)
+    else
+      replaceGlobsWithExistingDirectories(paths.flatMap(replaceFirstFoundWildcardWithDirectories))
+
+  private def pathMightBeValidResource(path: String): Boolean =
+    path.indexOf('*') == -1 &&
+      (path.endsWith(".zip") ||
+        path.endsWith(".dmg") ||
+        Paths.get(path.replace("file:", "")).isDirectory)
```

Comment:
why don't you construct path from URI properly instead of manual conversion. Also you are leaving trailing slashes now.